sia-antfarm/antfarm.go

```diff
@@ -0,0 +1,170 @@
+package main
+
+import (
+	"encoding/json"
+	"log"
+	"net"
+	"net/http"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/julienschmidt/httprouter"
+)
+
+type (
+	// AntfarmConfig contains the fields to parse and use to create a sia-antfarm.
+	AntfarmConfig struct {
+		ListenAddress string
+		AntConfigs    []AntConfig
+		AutoConnect   bool
+
+		// ExternalFarms is a slice of net addresses representing the API addresses
+		// of other ant farms to connect to.
+		ExternalFarms []string
+	}
+
+	// antFarm defines the 'antfarm' type. antFarm orchestrates a collection of
+	// ants and provides an API server to interact with them.
+	antFarm struct {
+		apiListener net.Listener
+		wg          sync.WaitGroup
+		ants        []*Ant
+		router      *httprouter.Router
+	}
+)
+
+// createAntfarm creates a new antFarm given the supplied AntfarmConfig
```

Comment:
documentation is inconsistent between antFarm, Antfarm: need to decide whether it's one word or two.